### PR TITLE
Perma-(De)Buff on zone Fix

### PR DESCRIPTION
**Pull request by chipmungus**
Added a function to status_effect_container, "delStatusEffectsByDuration"

Removes status effects which have a duration less than the time specified

Called the function in Scripts\Globals\Player.lua on zone, with a default time of 5 seconds.

Seems to have fixed issue #1461

### CODE REVIEW

File: src/map/status_effect_container.cpp [hunk 1]
```diff
@@ -564,6 +564,23 @@ void CStatusEffectContainer::DelStatusEffectsByType(uint16 Type)
...
+*																		*
+*  Removes Status Effects by Duration   						*
+*																		*
+************************************************************************/
+
+void CStatusEffectContainer::DelStatusEffectsByDuration(uint32 Duration)
+{
+	for (uint16 i = 0; i < m_StatusEffectList.size(); ++i)
+	{
+       if  (m_StatusEffectList.at(i)->GetDuration() <= Duration)
```
Comment (bendangelo) [inline src/map/status_effect_container.cpp]: Can you fix this formatting.

**Comment by bendangelo** [comment]
I'm kind of on the fence about this. I think the issue is status effects are not checked for removal while zoning. So on the next zone enter the status effect container should figure out the effects are past due. This leads me to believe this is an integer overflow bug.

**Comment by xdemolish** [comment]
dont need to make a new commit for each file changed

there's probably a better way to do this, i might look into it eventually

**Comment by teschnei** [comment]
Better to figure out why this is happening instead of band aiding it. Ie:
why are status effects becoming infinite duration? Are they being saved
with 0 duration? Is it the effect saving or effect loading that's doing it?

**Comment by teschnei** [comment]
Best place to start is where they are saved: does the server tell the
difference between an effect with 0 duration (infinite effect) and an
effect with 0 remaining duration (start time + duration = current tick)? I
could see it accidentally saving remaining time as 0 (which gets loaded as
infinite) when it should not be saving at all

**Comment by chipmungus** [comment]
Seems it could be dealt with by:

i, This "band-aid"

ii. Having the server save statuses with <1 duration as >1, bypassing the 0 = infinite thing when it loads after the zone.

iii. Remove "0 = infinite" as a rule. Not really sure why infinite duration status effects exist at all; there are none in retail that I'm aware of.

**Comment by teschnei** [comment]
Iv. The server must calculate the remaining duration before saving it. If
the remaining duration is 0 but the buff duration is not, don't save it

**Comment by TeoTwawki** [comment]
"Not really sure why infinite duration status effects exist at all; there are none in retail that I'm aware of."
Retail has multiple effects that do not wear off from time, but only by zoning/connection loss/death/whatever depending on which effect we are talking about. Farm yoursself up 4 days worth of Abyssea time, and go afk. Your ATMA will stay on the entire time.

**Comment by teschnei** [comment]
there are tons of infinite duration effects, and monsters use them sometimes as well

423ff9ec82ac724eaa70b44e4c1dd31c6668bc67

**Pull request closed without merging**